pkg/stripe/analytics_telemetry.go

```diff
@@ -0,0 +1,169 @@
+package stripe
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/go-querystring/query"
+	"github.com/google/uuid"
+
+	"github.com/spf13/cobra"
+
+	"github.com/stripe/stripe-cli/pkg/useragent"
+	"github.com/stripe/stripe-cli/pkg/version"
+)
+
+//
+// Public types
+//
+
+// TelemetryContextKey is the key for the telemetry context
+type TelemetryContextKey struct{}
+
+// TelemetryClientKey is the key for the telemetry client
+type TelemetryClientKey struct{}
+
+// DefaultTelemetryEndpoint is the default URL for the telemetry destination
+const DefaultTelemetryEndpoint = "https://r.stripe.com/0"
+
+// CLIAnalyticsEventContext is the structure that holds telemetry data context that is ultimately sent to the Stripe Analytics Service.
+type CLIAnalyticsEventContext struct {
```

Comment:
Context is an overloaded term here.. perhaps Metadata?

Reply:
True, updated